# Copyright (C) 2007  Miriam Ruiz <little_miry@yahoo.es>
# Distributed under the same license as the game. See debian/copyright.

Index: digiband-1.0.1/Source/drawtext.cpp
===================================================================
--- digiband-1.0.1.orig/Source/drawtext.cpp	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/drawtext.cpp	2007-09-14 22:29:02.000000000 +0000
@@ -634,11 +634,11 @@
 		}
 		fontfail=false;
 		char filename[256];
-		snprintf(filename,255,"Themes/%s/%s", theme,fontname);
+		snprintf(filename,255, DATADIR "/Themes/%s/%s", theme,fontname);
 		font=NULL;
 		font=TTF_OpenFont(filename, size);
 		if(!font) {
-			snprintf(filename,255,"Themes/Default/%s", fontname);
+			snprintf(filename,255, DATADIR "/Themes/Default/%s", fontname);
 			font=TTF_OpenFont(filename, size);
 			if(!font) {
 				printf("\nTTF_OpenFont: %s (%s)\n\n", TTF_GetError(),filename);
@@ -761,11 +761,11 @@
 		}
 		fontfail=false;
 		char filename[256];
-		snprintf(filename,255,"Themes/%s/%s", theme,fontname);
+		snprintf(filename,255, DATADIR "/Themes/%s/%s", theme,fontname);
 		font=NULL;
 		font=TTF_OpenFont(filename, size);
 		if(!font) {
-			snprintf(filename,255,"Themes/Default/%s", fontname);
+			snprintf(filename,255, DATADIR "/Themes/Default/%s", fontname);
 			font=TTF_OpenFont(filename, size);
 			if(!font) {
 				printf("\nTTF_OpenFont: %s\n\n", TTF_GetError());
@@ -934,13 +934,13 @@
 		}
 		fontfail=false;
 		char filename[256];
-		snprintf(filename,255,"Themes/%s/%s", theme,fontname);
+		snprintf(filename,255, DATADIR "/Themes/%s/%s", theme,fontname);
 		while(fstreampotty>0){}
 		fstreampotty++;
 		font=TTF_OpenFont(filename, size);
 		fstreampotty--;
 		if(!font) {
-			snprintf(filename,255,"Themes/Default/%s", fontname);
+			snprintf(filename,255, DATADIR "/Themes/Default/%s", fontname);
 			while(fstreampotty>0){}
 		    fstreampotty++;
 			font=TTF_OpenFont(filename, size);
Index: digiband-1.0.1/Source/playbgavi.cpp
===================================================================
--- digiband-1.0.1.orig/Source/playbgavi.cpp	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/playbgavi.cpp	2007-09-14 22:29:02.000000000 +0000
@@ -37,21 +37,21 @@
 		static bool logoisdefault=false;
         static char avifile[128];
         snprintf(avifile,128, "NULL");
-		if (currscreen==1||currscreen==2||currscreen==7||currscreen==8||currscreen==9||currscreen==10||currscreen==17)snprintf(avifile,127, "./Themes/%s/titlescreen.avi", theme);
-		if (currscreen==4||currscreen==5)snprintf(avifile,127, "./Themes/%s/songselection-bottom.avi", theme);
-		if (currscreen==6)snprintf(avifile,127, "./Themes/%s/gm.avi", theme);
-		if (currscreen==12)snprintf(avifile,127, "./Themes/%s/gameresult.avi", theme);
-		if (currscreen==13)snprintf(avifile,127, "./Themes/%s/ending.avi", theme);
+		if (currscreen==1||currscreen==2||currscreen==7||currscreen==8||currscreen==9||currscreen==10||currscreen==17)snprintf(avifile,127,  DATADIR "/Themes/%s/titlescreen.avi", theme);
+		if (currscreen==4||currscreen==5)snprintf(avifile,127,  DATADIR "/Themes/%s/songselection-bottom.avi", theme);
+		if (currscreen==6)snprintf(avifile,127,  DATADIR "/Themes/%s/gm.avi", theme);
+		if (currscreen==12)snprintf(avifile,127,  DATADIR "/Themes/%s/gameresult.avi", theme);
+		if (currscreen==13)snprintf(avifile,127,  DATADIR "/Themes/%s/ending.avi", theme);
 		if (currscreen==21){
-			if (!logoisdefault)snprintf(avifile,127, "./Themes/%s/logo.avi", theme);
-			if (logoisdefault)snprintf(avifile,127, "./Themes/Default/logo.avi", theme);
+			if (!logoisdefault)snprintf(avifile,127,  DATADIR "/Themes/%s/logo.avi", theme);
+			if (logoisdefault)snprintf(avifile,127,  DATADIR "/Themes/Default/logo.avi", theme);
 		}
 		if (!checked[currscreen]){
 			hasavi[currscreen]=testfile(avifile);
 			checked[currscreen]=true;
 			if (currscreen==21){
 				if (!hasavi[currscreen]){
-                	snprintf(avifile,127, "./Themes/Default/logo.avi", theme);
+                	snprintf(avifile,127,  DATADIR "/Themes/Default/logo.avi", theme);
                 	hasavi[currscreen]=testfile(avifile);
                 	logoisdefault=true;
 				}
Index: digiband-1.0.1/Source/playpreview.C
===================================================================
--- digiband-1.0.1.orig/Source/playpreview.C	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/playpreview.C	2007-09-14 23:07:52.000000000 +0000
@@ -34,36 +34,36 @@
 	switch(currscreen){
 		case 1:
 			if (firstboot){
-				snprintf(thepreview,255,"./Themes/%s/loading.mp3", theme);
+				snprintf(thepreview,255, DATADIR "/Themes/%s/loading.mp3", theme);
 				if(!testfile(thepreview))
-					snprintf(thepreview,255,"./Themes/Default/loading.mp3");
+					snprintf(thepreview,255, DATADIR "/Themes/Default/loading.mp3");
 				firstboot=false;
 			}else{
-				snprintf(thepreview,255,"./Themes/%s/titlescreen.mp3", theme);
+				snprintf(thepreview,255, DATADIR "/Themes/%s/titlescreen.mp3", theme);
 				if(!testfile(thepreview))
-					snprintf(thepreview,255,"./Themes/Default/titlescreen.mp3");
+					snprintf(thepreview,255, DATADIR "/Themes/Default/titlescreen.mp3");
 			}
 			break;
 		case 2:
-			snprintf(thepreview,255,"./Themes/%s/optionscreen.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/optionscreen.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/optionscreen.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/optionscreen.mp3");
 			break;
 		case 4:
-			snprintf(thepreview,255,"./Themes/%s/ModeSelection.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/ModeSelection.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/ModeSelection.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/ModeSelection.mp3");
 			break;
 		case 5:
 			if (!jukeboxmode){
-				snprintf(thepreview,255,"./Themes/%s/songselection.mp3", theme);
+				snprintf(thepreview,255, DATADIR "/Themes/%s/songselection.mp3", theme);
 				if(!testfile(thepreview))
-					snprintf(thepreview,255,"./Themes/Default/songselection.mp3");
+					snprintf(thepreview,255, DATADIR "/Themes/Default/songselection.mp3");
 				if (thesong>-1 && thesong<gbsongcount){
 					if (jukeboxmode){
 						bgmmp3.stop();
 					}else{
-						snprintf(thepreview,255,"./Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].previewwave);
+						snprintf(thepreview,255, DATADIR "/Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].previewwave);
 					}
 				}
 			}else{
@@ -74,19 +74,19 @@
 			snprintf(thepreview,255,"NULL");
 			break;
 		case 12:
-			snprintf(thepreview,255,"./Themes/%s/gameresults.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/gameresults.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/gameresults.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/gameresults.mp3");
 			break;
 		case 13:
-			snprintf(thepreview,255,"./Themes/%s/ending.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/ending.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/ending.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/ending.mp3");
 			break;
 		case 14:
-			snprintf(thepreview,255,"./Themes/%s/gameover.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/gameover.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/gameover.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/gameover.mp3");
 			break;
 		case 16:
 			snprintf(thepreview,255,"");
@@ -95,9 +95,9 @@
 			snprintf(thepreview,355,"NULL");
 			break;
 		case 21:
-			snprintf(thepreview,255,"./Themes/%s/logo.mp3", theme);
+			snprintf(thepreview,255, DATADIR "/Themes/%s/logo.mp3", theme);
 			if(!testfile(thepreview))
-				snprintf(thepreview,255,"./Themes/Default/logo.mp3");
+				snprintf(thepreview,255, DATADIR "/Themes/Default/logo.mp3");
 			break;
 	}
 	string error;
Index: digiband-1.0.1/Source/playsound.C
===================================================================
--- digiband-1.0.1.orig/Source/playsound.C	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/playsound.C	2007-09-14 22:29:02.000000000 +0000
@@ -55,9 +55,9 @@
 	if (strlen(soundfile)>1){
 		if (num<=256){
 			char thefile[256];
-			snprintf(thefile,255, "./Themes/%s/%s",theme,soundfile);
+			snprintf(thefile,255,  DATADIR "/Themes/%s/%s",theme,soundfile);
 			if (!testfile(thefile)){
-				snprintf(thefile,255, "./Themes/Default/%s",soundfile);
+				snprintf(thefile,255,  DATADIR "/Themes/Default/%s",soundfile);
 			}
             if (testfile(thefile)){
 				if (testxa(thefile)){
Index: digiband-1.0.1/Source/screeninit.C
===================================================================
--- digiband-1.0.1.orig/Source/screeninit.C	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/screeninit.C	2007-09-14 22:29:02.000000000 +0000
@@ -140,10 +140,10 @@
 	SDL_WM_SetCaption("Seijinohki DigiBand", "DIGIBAND");
 	SDL_Surface *theicon;
 	char icon[256];
-	snprintf(icon,255,"./Themes/%s/Icon.png",theme);
+	snprintf(icon,255, DATADIR "/Themes/%s/Icon.png",theme);
 	theicon=IMG_Load(icon);
 	if (theicon==NULL){
-		snprintf(icon,255, "./Themes/Default/Icon.png");
+		snprintf(icon,255,  DATADIR "/Themes/Default/Icon.png");
 		theicon=IMG_Load(icon);
 		if (theicon==NULL){
 			printf("Uh... I can't do that.\n");
@@ -289,10 +289,10 @@
 	SDL_WM_SetCaption("Seijinohki DigiBand", "DigiBand");
 	SDL_Surface *theicon;
 	char icon[256];
-	snprintf(icon,255,"./Themes/%s/Icon.png", theme);
+	snprintf(icon,255, DATADIR "/Themes/%s/Icon.png", theme);
 	theicon=IMG_Load(icon);
 	if (theicon==NULL){
-		snprintf(icon,255,"./Themes/Default/Icon.png");
+		snprintf(icon,255, DATADIR "/Themes/Default/Icon.png");
 		theicon=IMG_Load(icon);
 		if (theicon==NULL){
 			printf("Uh... I can't do that.");
Index: digiband-1.0.1/Source/song.cpp
===================================================================
--- digiband-1.0.1.orig/Source/song.cpp	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/song.cpp	2007-09-14 23:07:43.000000000 +0000
@@ -1476,22 +1476,22 @@
 		backgroundtexture=0;
 	}
 	if (thesong>-1&&thesong<gbsongcount){
-		snprintf(backgroundfile,255,"./Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].background);
-		snprintf(previewfile,255,"./Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].previewimg);
+		snprintf(backgroundfile,255, DATADIR "/Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].background);
+		snprintf(previewfile,255, DATADIR "/Songs/%s/%s/%s", songs[thesong].cat,songs[thesong].name,songs[thesong].previewimg);
 		backgroundtexture=loadtexture(backgroundfile);
 		if (backgroundtexture==0){
-			snprintf(backgroundfile,255, "./Songs/%s/%s/image.bmp", songs[thesong].cat,songs[thesong].name);
+			snprintf(backgroundfile,255, DATADIR "/Songs/%s/%s/image.bmp", songs[thesong].cat,songs[thesong].name);
 			backgroundtexture=loadtexture(backgroundfile);
 		}
 		if (backgroundtexture==0){
-			snprintf(backgroundfile,255, "./Songs/%s/%s/wall.jpg", songs[thesong].cat,songs[thesong].name);
+			snprintf(backgroundfile,255, DATADIR "/Songs/%s/%s/wall.jpg", songs[thesong].cat,songs[thesong].name);
 			backgroundtexture=loadtexture(backgroundfile);
 		}
 	}
 	if (backgroundtexture==0){
-		snprintf(backgroundfile,255, "./Themes/%s/wall.png", theme);
+		snprintf(backgroundfile,255,  DATADIR "/Themes/%s/wall.png", theme);
 		if (!testfile(backgroundfile)){
-			snprintf(backgroundfile,255, "./Themes/Default/wall.png");
+			snprintf(backgroundfile,255,  DATADIR "/Themes/Default/wall.png");
 			if (!testfile(backgroundfile)){
 				printf("Could not find any wall files.. Do you have a default theme (%s)?\n",theme);
 				printf("%s\n",songs[thesong].name);
@@ -1503,22 +1503,22 @@
 	previewtexture=loadtexture(previewfile);
 	if (thesong>-1&&thesong<gbsongcount){
 		if (previewtexture==0){
-			snprintf(previewfile,255, "./Songs/%s/%s/q.png", songs[thesong].cat,songs[thesong].name);
+			snprintf(previewfile,255, DATADIR "/Songs/%s/%s/q.png", songs[thesong].cat,songs[thesong].name);
 			previewtexture=loadtexture(previewfile);
 		}
 		if (previewtexture==0){
-			snprintf(previewfile,255, "./Songs/%s/%s/q.jpg", songs[thesong].cat,songs[thesong].name);
+			snprintf(previewfile,255, DATADIR "/Songs/%s/%s/q.jpg", songs[thesong].cat,songs[thesong].name);
 			previewtexture=loadtexture(previewfile);
 		}
 		if (previewtexture==0){
-			snprintf(previewfile,255, "./Songs/%s/%s/pre.jpg", songs[thesong].cat,songs[thesong].name);
+			snprintf(previewfile,255, DATADIR "/Songs/%s/%s/pre.jpg", songs[thesong].cat,songs[thesong].name);
 			previewtexture=loadtexture(previewfile);
 		}
 	}
 	if (previewtexture==0){
-		snprintf(previewfile,255, "./Themes/%s/q.png", theme);
+		snprintf(previewfile,255,  DATADIR "/Themes/%s/q.png", theme);
 		if (!testfile(previewfile)){
-			snprintf(previewfile,255, "./Themes/Default/q.png");
+			snprintf(previewfile,255,  DATADIR "/Themes/Default/q.png");
 			printf("error: can't find default theme file q.png\n");
 			appisrunning=false;
 		}
Index: digiband-1.0.1/Source/systemload.C
===================================================================
--- digiband-1.0.1.orig/Source/systemload.C	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/systemload.C	2007-09-14 23:01:55.000000000 +0000
@@ -35,7 +35,7 @@
 		texani[a][3]=0.0;
 		snprintf(texstoload[a],255,"");
 	}
-	snprintf(themedir,255,"./Themes/%s", theme);
+	snprintf(themedir,255, DATADIR "/Themes/%s", theme);
 	d=opendir(themedir);
 	if (d==NULL){
 		printf("Security settings for %s Themes directory incorrect.\n", theme);
@@ -53,7 +53,7 @@
 			if (strcmp(strlwr2(temp),"statusbar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -66,7 +66,7 @@
 			if (strcmp(strlwr2(temp),"titlescreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -79,7 +79,7 @@
 			if (strcmp(strlwr2(temp),"titlescreenarcade")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -92,7 +92,7 @@
 			if (strcmp(strlwr2(temp),"optionsscreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -105,7 +105,7 @@
 			if (strcmp(strlwr2(temp),"selection")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -118,7 +118,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-bg")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -131,7 +131,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-drum")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -144,7 +144,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-guitar1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -157,7 +157,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-guitar2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -170,7 +170,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-bottom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -183,7 +183,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-top")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -196,7 +196,7 @@
 			if (strcmp(strlwr2(temp),"songselection-bottom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -209,7 +209,7 @@
 			if (strcmp(strlwr2(temp),"songselection-selected")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -222,7 +222,7 @@
 			if (strcmp(strlwr2(temp),"songselection-deselected")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -235,7 +235,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumguitlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -248,7 +248,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumguitlayerwide")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -261,7 +261,7 @@
 			if (strcmp(strlwr2(temp),"diffselect")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -274,7 +274,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -287,7 +287,7 @@
 			if (strcmp(strlwr2(temp),"gm-guitlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -300,7 +300,7 @@
 			if (strcmp(strlwr2(temp),"perfect")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -313,7 +313,7 @@
 			if (strcmp(strlwr2(temp),"great")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -326,7 +326,7 @@
 			if (strcmp(strlwr2(temp),"good")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -339,7 +339,7 @@
 			if (strcmp(strlwr2(temp),"almost")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -352,7 +352,7 @@
 			if (strcmp(strlwr2(temp),"miss")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -365,7 +365,7 @@
 			if (strcmp(strlwr2(temp),"scorenum")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -378,7 +378,7 @@
 			if (strcmp(strlwr2(temp),"combo")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -391,7 +391,7 @@
 			if (strcmp(strlwr2(temp),"gameover-l")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -404,7 +404,7 @@
 			if (strcmp(strlwr2(temp),"gameover-r")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -417,7 +417,7 @@
 			if (strcmp(strlwr2(temp),"gameresult1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -430,7 +430,7 @@
 			if (strcmp(strlwr2(temp),"gameresult2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -443,7 +443,7 @@
 			if (strcmp(strlwr2(temp),"particle")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -456,7 +456,7 @@
 			if (strcmp(strlwr2(temp),"white")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -469,7 +469,7 @@
 			if (strcmp(strlwr2(temp),"scrollbar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -482,7 +482,7 @@
 			if (strcmp(strlwr2(temp),"scrollbox")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -495,7 +495,7 @@
 			if (strcmp(strlwr2(temp),"gametitle")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -508,7 +508,7 @@
 			if (strcmp(strlwr2(temp),"danger")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -521,7 +521,7 @@
 			if (strcmp(strlwr2(temp),"death")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -534,7 +534,7 @@
 			if (strcmp(strlwr2(temp),"timer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -547,7 +547,7 @@
 			if (strcmp(strlwr2(temp),"autoplay")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -560,7 +560,7 @@
 			if (strcmp(strlwr2(temp),"longver")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -573,7 +573,7 @@
 			if (strcmp(strlwr2(temp),"gm-backlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -586,7 +586,7 @@
 			if (strcmp(strlwr2(temp),"logobg")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -599,7 +599,7 @@
 			if (strcmp(strlwr2(temp),"openal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -612,7 +612,7 @@
 			if (strcmp(strlwr2(temp),"opengl")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/%s/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/%s/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -621,7 +621,7 @@
 			}
 		}
 	}
-	snprintf(themedir,255,"./Noteskins/%s", noteskin);
+	snprintf(themedir,255, DATADIR "/Noteskins/%s", noteskin);
 	d=opendir(themedir);
 	if (d==NULL){
 		printf("Security settings for Noteskins directory incorrect.\n");
@@ -636,7 +636,7 @@
 			if (strcmp(strlwr2(temp),"drumtrack")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -649,7 +649,7 @@
 			if (strcmp(strlwr2(temp),"hihat")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -662,7 +662,7 @@
 			if (strcmp(strlwr2(temp),"snare")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -675,7 +675,7 @@
 			if (strcmp(strlwr2(temp),"hitom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -688,7 +688,7 @@
 			if (strcmp(strlwr2(temp),"lotom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -701,7 +701,7 @@
 			if (strcmp(strlwr2(temp),"cymbal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -714,7 +714,7 @@
 			if (strcmp(strlwr2(temp),"note1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -727,7 +727,7 @@
 			if (strcmp(strlwr2(temp),"note2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -740,7 +740,7 @@
 			if (strcmp(strlwr2(temp),"note3")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -753,7 +753,7 @@
 			if (strcmp(strlwr2(temp),"note4")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -766,7 +766,7 @@
 			if (strcmp(strlwr2(temp),"note5")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -779,7 +779,7 @@
 			if (strcmp(strlwr2(temp),"note6")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -792,7 +792,7 @@
 			if (strcmp(strlwr2(temp),"bass")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -805,7 +805,7 @@
 			if (strcmp(strlwr2(temp),"guittrack")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -818,7 +818,7 @@
 			if (strcmp(strlwr2(temp),"lifebar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -831,7 +831,7 @@
 			if (strcmp(strlwr2(temp),"scorebar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -844,7 +844,7 @@
 			if (strcmp(strlwr2(temp),"measure")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -857,7 +857,7 @@
 			if (strcmp(strlwr2(temp),"quarter")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -870,7 +870,7 @@
 			if (strcmp(strlwr2(temp),"wail")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -883,7 +883,7 @@
 			if (strcmp(strlwr2(temp),"gred")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -896,7 +896,7 @@
 			if (strcmp(strlwr2(temp),"ggreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -909,7 +909,7 @@
 			if (strcmp(strlwr2(temp),"gblue")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -922,7 +922,7 @@
 			if (strcmp(strlwr2(temp),"explosion")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -935,7 +935,7 @@
 			if (strcmp(strlwr2(temp),"petal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -948,7 +948,7 @@
 			if (strcmp(strlwr2(temp),"modifier")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/%s/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/%s/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -957,7 +957,7 @@
 			}
 		}
 	}
-	snprintf(themedir,255,"./Themes/Default");
+	snprintf(themedir,255, DATADIR "/Themes/Default");
 	d=opendir(themedir);
 	if (d==NULL){
 		printf("Security settings for Themes directory incorrect.\n");
@@ -973,7 +973,7 @@
 			if (strcmp(strlwr2(temp),"statusbar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -986,7 +986,7 @@
 			if (strcmp(strlwr2(temp),"titlescreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -999,7 +999,7 @@
 			if (strcmp(strlwr2(temp),"titlescreenarcade")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1012,7 +1012,7 @@
 			if (strcmp(strlwr2(temp),"optionsscreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1025,7 +1025,7 @@
 			if (strcmp(strlwr2(temp),"selection")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1038,7 +1038,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-bg")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1051,7 +1051,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-drum")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1064,7 +1064,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-guitar1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1077,7 +1077,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-guitar2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1090,7 +1090,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-bottom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1103,7 +1103,7 @@
 			if (strcmp(strlwr2(temp),"modeselection-top")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1116,7 +1116,7 @@
 			if (strcmp(strlwr2(temp),"songselection-bottom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1129,7 +1129,7 @@
 			if (strcmp(strlwr2(temp),"songselection-selected")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1142,7 +1142,7 @@
 			if (strcmp(strlwr2(temp),"songselection-deselected")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1155,7 +1155,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumguitlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1168,7 +1168,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumguitlayerwide")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1181,7 +1181,7 @@
 			if (strcmp(strlwr2(temp),"diffselect")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1194,7 +1194,7 @@
 			if (strcmp(strlwr2(temp),"gm-drumlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1207,7 +1207,7 @@
 			if (strcmp(strlwr2(temp),"gm-guitlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1220,7 +1220,7 @@
 			if (strcmp(strlwr2(temp),"perfect")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1233,7 +1233,7 @@
 			if (strcmp(strlwr2(temp),"great")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1246,7 +1246,7 @@
 			if (strcmp(strlwr2(temp),"good")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1259,7 +1259,7 @@
 			if (strcmp(strlwr2(temp),"almost")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1272,7 +1272,7 @@
 			if (strcmp(strlwr2(temp),"miss")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1285,7 +1285,7 @@
 			if (strcmp(strlwr2(temp),"scorenum")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1298,7 +1298,7 @@
 			if (strcmp(strlwr2(temp),"combo")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1311,7 +1311,7 @@
 			if (strcmp(strlwr2(temp),"gameover-l")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1324,7 +1324,7 @@
 			if (strcmp(strlwr2(temp),"gameover-r")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1337,7 +1337,7 @@
 			if (strcmp(strlwr2(temp),"gameresult1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1350,7 +1350,7 @@
 			if (strcmp(strlwr2(temp),"gameresult2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1363,7 +1363,7 @@
 			if (strcmp(strlwr2(temp),"particle")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1376,7 +1376,7 @@
 			if (strcmp(strlwr2(temp),"white")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1389,7 +1389,7 @@
 			if (strcmp(strlwr2(temp),"scrollbar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1402,7 +1402,7 @@
 			if (strcmp(strlwr2(temp),"scrollbox")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1415,7 +1415,7 @@
 			if (strcmp(strlwr2(temp),"gametitle")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1428,7 +1428,7 @@
 			if (strcmp(strlwr2(temp),"danger")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1441,7 +1441,7 @@
 			if (strcmp(strlwr2(temp),"death")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1454,7 +1454,7 @@
 			if (strcmp(strlwr2(temp),"timer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1467,7 +1467,7 @@
 			if (strcmp(strlwr2(temp),"autoplay")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1480,7 +1480,7 @@
 			if (strcmp(strlwr2(temp),"longver")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1493,7 +1493,7 @@
 			if (strcmp(strlwr2(temp),"gm-backlayer")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1506,7 +1506,7 @@
 			if (strcmp(strlwr2(temp),"logobg")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1519,7 +1519,7 @@
 			if (strcmp(strlwr2(temp),"openal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1532,7 +1532,7 @@
 			if (strcmp(strlwr2(temp),"opengl")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Themes/Default/%s", theme,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Themes/Default/%s", theme,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1541,7 +1541,7 @@
 			}
 		}
 	}
-	snprintf(themedir,255,"./Noteskins/Default");
+	snprintf(themedir,255, DATADIR "/Noteskins/Default");
 	d=opendir(themedir);
 	if (d==NULL){
 		printf("Security settings for Noteskins directory incorrect.\n");
@@ -1557,7 +1557,7 @@
 			if (strcmp(strlwr2(temp),"drumtrack")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1570,7 +1570,7 @@
 			if (strcmp(strlwr2(temp),"hihat")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1583,7 +1583,7 @@
 			if (strcmp(strlwr2(temp),"snare")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1596,7 +1596,7 @@
 			if (strcmp(strlwr2(temp),"hitom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1609,7 +1609,7 @@
 			if (strcmp(strlwr2(temp),"lotom")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1622,7 +1622,7 @@
 			if (strcmp(strlwr2(temp),"cymbal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1635,7 +1635,7 @@
 			if (strcmp(strlwr2(temp),"note1")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1648,7 +1648,7 @@
 			if (strcmp(strlwr2(temp),"note2")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1661,7 +1661,7 @@
 			if (strcmp(strlwr2(temp),"note3")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1674,7 +1674,7 @@
 			if (strcmp(strlwr2(temp),"note4")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1687,7 +1687,7 @@
 			if (strcmp(strlwr2(temp),"note5")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1700,7 +1700,7 @@
 			if (strcmp(strlwr2(temp),"note6")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1713,7 +1713,7 @@
 			if (strcmp(strlwr2(temp),"bass")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1726,7 +1726,7 @@
 			if (strcmp(strlwr2(temp),"guittrack")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1739,7 +1739,7 @@
 			if (strcmp(strlwr2(temp),"lifebar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1752,7 +1752,7 @@
 			if (strcmp(strlwr2(temp),"scorebar")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1765,7 +1765,7 @@
 			if (strcmp(strlwr2(temp),"measure")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1778,7 +1778,7 @@
 			if (strcmp(strlwr2(temp),"quarter")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1791,7 +1791,7 @@
 			if (strcmp(strlwr2(temp),"wail")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1804,7 +1804,7 @@
 			if (strcmp(strlwr2(temp),"gred")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1817,7 +1817,7 @@
 			if (strcmp(strlwr2(temp),"ggreen")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1830,7 +1830,7 @@
 			if (strcmp(strlwr2(temp),"gblue")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1843,7 +1843,7 @@
 			if (strcmp(strlwr2(temp),"explosion")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1856,7 +1856,7 @@
 			if (strcmp(strlwr2(temp),"petal")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1869,7 +1869,7 @@
 			if (strcmp(strlwr2(temp),"modifier")==0){
 				if (strlen(texstoload[t])<2){
 					if (dir->d_name[s]=='-'){
-						snprintf(texstoload[t],255,"./Noteskins/Default/%s", noteskin,dir->d_name);
+						snprintf(texstoload[t],255, DATADIR "/Noteskins/Default/%s", noteskin,dir->d_name);
 						texani[t][0]=double(dbstoi(strtok((dir->d_name+s+1),"x")));
 						texani[t][1]=double(dbstoi(strtok(NULL,".")));
 						//printf("found data: %s (%fx%f) #%i\n",texstoload[t],texani[t][0],texani[t][1],t);
@@ -1903,9 +1903,9 @@
 	cachesongs();
 	fadout();
 	char previewfile[128];
-	snprintf(previewfile,127,"./Themes/%s/q.png", theme);
+	snprintf(previewfile,127, DATADIR "/Themes/%s/q.png", theme);
 	if (!testfile(previewfile)){
-		snprintf(previewfile,127,"./Themes/Default/q.png");
+		snprintf(previewfile,127, DATADIR "/Themes/Default/q.png");
 	}
 	previewtexture=loadtexture(previewfile);
 }
@@ -1927,15 +1927,15 @@
 	if (currscreen==5)genlist();
 	
 	char previewfile[128];
-	snprintf(previewfile,127,"./Themes/%s/q.png", theme);
+	snprintf(previewfile,127, DATADIR "/Themes/%s/q.png", theme);
 	if (!testfile(previewfile)){
-		snprintf(previewfile,127,"./Themes/Default/q.png");
+		snprintf(previewfile,127, DATADIR "/Themes/Default/q.png");
 	}
 	previewtexture=loadtexture(previewfile);
 	char backgroundfile[128];
-	snprintf(backgroundfile,127,"./Themes/%s/wall.png", theme);
+	snprintf(backgroundfile,127, DATADIR "/Themes/%s/wall.png", theme);
 	if (!testfile(backgroundfile)){
-		snprintf(backgroundfile,127,"./Themes/Default/wall.png");
+		snprintf(backgroundfile,127, DATADIR "/Themes/Default/wall.png");
 	}
 	backgroundtexture=loadtexture(backgroundfile);
 }
Index: digiband-1.0.1/Source/themeini.cpp
===================================================================
--- digiband-1.0.1.orig/Source/themeini.cpp	2007-09-14 22:26:21.000000000 +0000
+++ digiband-1.0.1/Source/themeini.cpp	2007-09-14 22:29:02.000000000 +0000
@@ -180,8 +180,8 @@
 	ifstream themefile[2];
 	char file[256];
 	for (int a=0;a<2;a++){
-		if (a==0)snprintf(file,255,"./Themes/Default/theme.ini");
-		if (a==1)snprintf(file,255,"./Themes/%s/theme.ini", theme);
+		if (a==0)snprintf(file,255, DATADIR "/Themes/Default/theme.ini");
+		if (a==1)snprintf(file,255, DATADIR "/Themes/%s/theme.ini", theme);
 		if (!testfile(file)){
 			printf("%s theme file not open, cannot find or other error.\n",file);
 			if (a==0)appisrunning=false;
Index: digiband-1.0.1/Source/cachesongs.C
===================================================================
--- digiband-1.0.1.orig/Source/cachesongs.C	2007-09-14 22:47:49.000000000 +0000
+++ digiband-1.0.1/Source/cachesongs.C	2007-09-14 23:08:34.000000000 +0000
@@ -43,7 +43,7 @@
 	}
 	
 	a=0;
-	d=opendir("./Noteskins");
+	d=opendir(DATADIR "/Noteskins");
 	if (d==NULL){
 		printf("Security settings for noteskin directory incorrect.");
 		exit(1);
@@ -99,12 +99,12 @@
 		}
 		struct stat idiots;
 		char fuckingidiots[1024];
-		snprintf(fuckingidiots,1023, "./Songs/%s", dir->d_name);
+		snprintf(fuckingidiots,1023, DATADIR "/Songs/%s", dir->d_name);
 		stat(fuckingidiots,&idiots);
 		if (!S_ISDIR(idiots.st_mode)){
 			continue;
 		}
-		snprintf(dircat,255, "./Songs/%s", dir->d_name);
+		snprintf(dircat,255, DATADIR "/Songs/%s", dir->d_name);
 		snprintf(catlist[catcount],255, "%s", dir->d_name);
 		d2=opendir(dircat);
 		while(cat=readdir(d2)){
@@ -114,7 +114,7 @@
 			}
 			struct stat grumblegrumble;
 			char fuckinggrr[1024];
-			snprintf (fuckinggrr,1023,"./Songs/%s/%s",dir->d_name,cat->d_name);
+			snprintf (fuckinggrr,1023, DATADIR "/Songs/%s/%s",dir->d_name,cat->d_name);
 			stat(fuckinggrr,&grumblegrumble);
 			if (!S_ISDIR(grumblegrumble.st_mode)){
 				continue;
@@ -127,7 +127,7 @@
 			catloading.load(dir->d_name,"font.ttf",255,255,255,24);
 			catloading.draw(0,50,0);
 			nextobject();
-			snprintf(songdir,511, "./Songs/%s/%s/", dir->d_name,cat->d_name);
+			snprintf(songdir,511, DATADIR "/Songs/%s/%s/", dir->d_name,cat->d_name);
 			cachesong(songdir,cat->d_name,dir->d_name);
 		}
 		catcount++;
Index: digiband-1.0.1/Source/movenext.C
===================================================================
--- digiband-1.0.1.orig/Source/movenext.C	2007-09-14 23:05:58.000000000 +0000
+++ digiband-1.0.1/Source/movenext.C	2007-09-14 23:08:10.000000000 +0000
@@ -44,12 +44,12 @@
 		}
 		struct stat idiots;
 		char fuckingidiots[1024];
-		snprintf(fuckingidiots,1023, "./Songs/%s", dir->d_name);
+		snprintf(fuckingidiots,1023, DATADIR "/Songs/%s", dir->d_name);
 		stat(fuckingidiots,&idiots);
 		if (!S_ISDIR(idiots.st_mode)){
 			continue;
 		}
-		snprintf(dircat,255,"./Songs/%s",dir->d_name);
+		snprintf(dircat,255, DATADIR "/Songs/%s",dir->d_name);
 		d2=opendir(dircat);
 		catcount++;
 		while(cat=readdir(d2)){
@@ -58,7 +58,7 @@
 			}
 			struct stat grumblegrumble;
 			char fuckinggrr[1024];
-			snprintf (fuckinggrr,1023,"./Songs/%s/%s",dir->d_name,cat->d_name);
+			snprintf (fuckinggrr,1023, DATADIR "/Songs/%s/%s",dir->d_name,cat->d_name);
 			stat(fuckinggrr,&grumblegrumble);
 			if (!S_ISDIR(grumblegrumble.st_mode)){
 				continue;
